### ISSUE TITLE
[no-useless-constructor] <Still catching error>

### ISSUE THREAD

**Issue by Mister-Hope** [typescript-eslint/typescript-eslint#1443]
After a year, we still got problems like [issue#15](https://github.com/typescript-eslint/typescript-eslint/issues/15)

**Repro**

```yml
extends:
  - eslint:all
  - plugin:@typescript-eslint/eslint-recommended
  - "plugin:@typescript-eslint/recommended"

parser: "vue-eslint-parser"

parserOptions:
  parser: "@typescript-eslint/parser"

```

```TS
// your repro code case
/* eslint-disable */

declare module 'aaa' {
  class Logger {
    options: any;
    constructor(options?: any);
  }

  const logger: Logger;
  export = logger;
}
```

**Expected Result**

No error

**Actual Result**

Eslint crush

**Additional Info**

```
[Error - 下午9:38:15] ESLint stack trace:
[Error - 下午9:38:15] TypeError: Cannot read property 'body' of null
Occurred while linting Z:\npm\vuepress-theme-hope\packages\theme-types\index.d.ts:24
    at checkForConstructor (Z:\npm\vuepress-theme-hope\node_modules\eslint\lib\rules\no-useless-constructor.js:161:42)
    at Z:\npm\vuepress-theme-hope\node_modules\eslint\lib\linter\safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (Z:\npm\vuepress-theme-hope\node_modules\eslint\lib\linter\safe-emitter.js:45:38)
    at NodeEventGenerator.applySelector (Z:\npm\vuepress-theme-hope\node_modules\eslint\lib\linter\node-event-generator.js:254:26)
    at NodeEventGenerator.applySelectors (Z:\npm\vuepress-theme-hope\node_modules\eslint\lib\linter\node-event-generator.js:283:22)
    at NodeEventGenerator.enterNode (Z:\npm\vuepress-theme-hope\node_modules\eslint\lib\linter\node-event-generator.js:297:14)
    at CodePathAnalyzer.enterNode (Z:\npm\vuepress-theme-hope\node_modules\eslint\lib\linter\code-path-analysis\code-path-analyzer.js:634:23)
    at Z:\npm\vuepress-theme-hope\node_modules\eslint\lib\linter\linter.js:936:32
    at Array.forEach (<anonymous>)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.15.0` |
| `@typescript-eslint/parser`        | `2.15.0` |
| `TypeScript`                       | `3.7.4` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.14.1` |
| `yarn`                              | `1.21.1` |

**Comment by armano2** [comment]
for unknown reason you are using core `no-useless-constructor` instead of our `@typescript-eslint/no-useless-constructor` rule

**Comment by armano2** [comment]
looks like this rule is missing in config:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.json

**Comment by Mister-Hope** [comment]
> for unknown reason you are using core `no-useless-constructor` instead of our `@typescript-eslint/no-useless-constructor` rule

Sorry, do you mean I have to disable `no-useless-constructor`?

I did not see it in [here](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md)

**Comment by Mister-Hope** [comment]
> looks like this rule is missing in config:
> https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.json

Got it, hoping you can add it in the next release, I'll disable it for workaround now. 😃

**Comment by armano2** [comment]
yes you should disable `no-useless-constructor` and enable `@typescript-eslint/no-useless-constructor`

**Comment by bradzacher** [comment]
> looks like this rule is missing in config:
> [/packages/eslint-plugin/src/configs/recommended.json@`master`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/recommended.json) [](/typescript-eslint/typescript-eslint/blob/HEAD@{2020-01-13T13:58:57Z}/packages/eslint-plugin/src/configs/recommended.json)

`no-useless-constructor` is not a recommended eslint base rule, so it won't be included in the recommended set.

This problem is because their config is using `eslint:all`, which turns on this rule. We don't have an `eslint-all` config.

We should instead update the documentation to include information about extension rules. The FAQ would be a good spot.

Closing as there's nothing to explicitly fix here.
